aws/resource_aws_codebuild_project.go

```diff
@@ -565,6 +570,8 @@ func expandProjectArtifactData(data map[string]interface{}) codebuild.ProjectArt
 		projectArtifacts.EncryptionDisabled = aws.Bool(data["encryption_disabled"].(bool))
 	}
 
+	projectArtifacts.OverrideArtifactName = aws.Bool(data["override_artifact_name"].(bool))
```

Comment:
Thanks for fixing. Just wondering for the future: my assumption was that `override_artifact_name` has a default value and therefore cannot be `nil`?

Reply:
Since there were two separate configuration block schemas (`artifacts` which had `override_artifact_name` declared and `secondary_artifacts` which did not have `override_artifact_name` declared), you're correct that the `Default` meant that the map key was always present when working with `artifacts`, but it was the `secondary_artifacts` map that did not have the `override_artifact_name` key since it was not declared in the schema. 👍

Reply:
We always recommend running the acceptance testing for the whole resource to tease out issues like these 😄

Reply:
Thanks for explaining. I just realized that updates for `secondary_artifacts` are not working correctly yet, so I added some more tests to cover this and tried to provide a fix: https://github.com/terraform-providers/terraform-provider-aws/pull/9652